test-data/unit/check-kwargs.test

```diff
@@ -86,6 +86,35 @@ def f(a: 'A') -> None: pass # N: "f" defined here
 f(b=object()) # E: Unexpected keyword argument "b" for "f"
 class A: pass
 
+[case testKeywordMisspelling]
+def f(other: 'A') -> None: pass # N: "f" defined here
+f(otter=A()) # E: Unexpected keyword argument "otter" for "f";  did you mean "other"?
+class A: pass
+
+[case testMultipleKeywordsForMisspelling]
+def f(thing : 'A', other: 'A', atter: 'A', btter: 'B') -> None: pass # N: "f" defined here
+f(otter=A()) # E: Unexpected keyword argument "otter" for "f";  did you mean "other" or "atter"?
+class A: pass
+class B: pass
+
+[case testKeywordMisspellingDifferentType]
+def f(other: 'A') -> None: pass # N: "f" defined here
+f(otter=B()) # E: Unexpected keyword argument "otter" for "f";  did you mean "other"?
+class A: pass
+class B: pass
+
+[case testKeywordMisspellingInheritance]
+def f(atter: 'A', btter: 'B', ctter: 'C') -> None: pass # N: "f" defined here
+f(otter=B()) # E: Unexpected keyword argument "otter" for "f";  did you mean "btter" or "atter"?
+class A: pass
+class B(A): pass
+class C: pass
+
+[case testKeywordMisspellingFloatInt]
+def f(atter: float, btter: int) -> None: pass # N: "f" defined here
+x: int = 5
+f(otter=x) # E: Unexpected keyword argument "otter" for "f";  did you mean "btter" or "atter"?
```

Comment:
Could you please add a test or two involving `*args` to be sure they don't interfere with the logic?

Reply:
Added some tests. 
And the *args did interfere with the logic. --> improved logic